components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -2807,6 +2808,22 @@ public static APIStore getExternalAPIStore(String apiStoreName, int tenantId) th
         }
     }
 
+    public static OrganizationDTO getOrganizationDTOFromTenantDomain(String requestedTenantDomain) {
+        OrganizationDTO organizationDTO = new OrganizationDTO(requestedTenantDomain);
+        organizationDTO.setRequestedTenantDomain(requestedTenantDomain);
+        return  organizationDTO;
+    }
+
+    public static OrganizationDTO getOrganizationDTOFromOrgID(String orgId, String requestedTenantDomain) {
+        OrganizationDTO organizationDTO = new OrganizationDTO(orgId);
+        if (requestedTenantDomain != null) {
+            organizationDTO.setRequestedTenantDomain(requestedTenantDomain);
+        } else {
+            organizationDTO.setRequestedTenantDomain(CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+        }
+        return  organizationDTO;
+    }
```

Comment:
```suggestion
    public static OrganizationDTO getOrganizationDTOFromOrgID(String orgId) {
        OrganizationDTO organizationDTO = new OrganizationDTO(orgId);
        organizationDTO.setRequestedTenantDomain(CARBON.SUPER);
        return  organizationDTO;
    }
```

Reply:
We can't assign **CARBON.SUPER** to requested tenantDomain. This method is called by Dev-Portal Rest APIs too as well as in all on-prem workflows. In Dev-Portal requests, if a user requests for a tenant domain we need to pass that to be set as the requestedTenantDomain attribute.